### DISCUSSION TITLE
Are aliases supposed to work in `(Alias)QualifiedName`s?

### DISCUSSION THREAD

**Discussion by DoctorKrolic** [dotnet/roslyn#67368] [Q&A] [answered]
Take a look at the following code:
```cs
using MySystem = System;

var emptyString = global::MySystem.String.Empty;
```
As of now, it doesn't compile: https://sharplab.io/#v2:EYLgtghglgdgNAFxANwKYCcoDMCecAmIA1AD4ACADAAQCyOZAjFQLxWMDcAsAFA/IToqqMAAcEOAMoJMMAOYsqsgDYB7YBCUgQdRgDopM2boCio8eyA=

However, I am not sure if it is a bug or an expected behaviour. From one point of view alias is just a placeholder for the thing it understudies, so `global::MySystem.String.Empty` should be equal to `global::System.String.Empty`, which is legal. But from the other point alias is only source-level thing, scoped to a file, meaning here `MySystem` is not a member of `global` namespace. What is a correct answer in such case?

**Comment by CyrusNajmabadi** [comment, marked as answer]
Compiler behavior seems to be correct here.  The point of an alias is to look up something in the global namespace of that set of references.  The set of references has no 'MySystem', so it errors.  The 'MySystem' alias isn't something found in something else.  It's for simple name lookup within it's lexical scope.  You don't have a simple name here, so it shouldn't even be referenced.

> From one point of view alias is just a placeholder for the thing it understudies

Definitely not how the language sees it :)

It's not like there's a prepass replacing that identifier with the thing on the RHS.  Instead, names are looked up.  If they happen to bind to the alias using that lookup logic, at that point you can consider it to have the meaning of the thing on the RHS.

**Reply by CyrusNajmabadi** [reply]
SImilarly, if you had this:

```c#
namespace X
{
    using A = Y;
    namespace Y
    {
        class C
        {
             private X.A.C c; // <-- not legal
        }
    }
}
```

'A' is not a member of 'X', so it is not found in `X.A`.  Instead, It's just an item available during simple name lookup.  The AliasQualified case is the same here.
